src/mpi4py/MPI/asdlpack.pxi

```diff
@@ -168,18 +170,14 @@ cdef int Py_GetDLPackBuffer(object obj, Py_buffer *view, int flags) except -1:
     cdef bint fixnull
 
     try:
-        dlpack = obj.__dlpack__
+        # we check __dlpack_device__ first instead of __dlpack__ to avoid
```

Comment:
I don't like this implementation of yours. Mine was very careful of avoiding side-effects (e.g. do not lookup and call methods at the same time). Moreover, in my implementation the `__dlpack_device__` method is optional.

Reply:
This is now reverted.